orc8r/cloud/go/services/certifier/analytics/calculations/lifespan.go

```diff
@@ -82,12 +83,14 @@ func calculateCertLifespanHours(certsDirectory string, certName string, metricCo
 func getCert(certPath string) ([]byte, error) {
 	dat, err := ioutil.ReadFile(certPath)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, fmt.Sprintf("failed to read cert file %s", certPath))
 	}
 	if strings.HasSuffix(certPath, ".pem") || strings.HasSuffix(certPath, ".crt") {
 		block, _ := pem.Decode(dat)
-		if block == nil || block.Type != "PUBLIC KEY" {
-			log.Fatal("failed to decode PEM block containing public key")
+		if block == nil {
+			return nil, fmt.Errorf("failed to decode a PEM block containing public key for certificate %s", certPath)
+		} else if block.Type != "PUBLIC KEY" && block.Type != "CERTIFICATE" {
+			return nil, fmt.Errorf("certificate %s has a PEM block type that is not PUBLIC KEY or CERTIFICATE, and instead is %s", certPath, block.Type)
```

Comment:
@andreilee Can you help me understand this better. Why are we unable to decode these certs and determine the expiration period ?

Reply:
We're expecting these types - `PRIVATE KEY` or a CSR would indicate a misuse I think.